diffpy/pdfmorph/pdfmorphapp.py

```diff
@@ -412,7 +468,178 @@ def main():
+        except KeyError:
+            if opts.serfile is not None:
+                parser.custom_error("The requested field was not found in the metadata file.")
+            else:
+                parser.custom_error("The requested field is missing from a PDF file header.")
+    else:
+        # Default is alphabetical sort
+        target_list.sort(reverse=opts.reverse)
+
+    # Disable single morph plotting
+    plot_opt = opts.plot
+    opts.plot = False
+
+    # Manage saving
+    save_opt = opts.saveloc
+    save_names = {}
+    save_morphs_here = ""
+    if save_opt is not None:
+        try:
+            # Make directory to save files in if it does not already exist
+            Path(save_opt).mkdir(parents=True, exist_ok=True)
+
+            # Morphs will be saved in the subdirectory "Morphs"
+            save_morphs_here = Path(save_opt).joinpath("Morphs")
+            save_morphs_here.mkdir(exist_ok=True)
+
+            # Get names for the saved morphs
+            if opts.snfile is not None:
+                # Names should be stored properly in opts.snfile
+                save_names = tools.deserialize(opts.snfile)
+            # Default naming scheme
+            else:
+                for target_file in target_list:
+                    save_names.update({target_file.name: {"save_morph_as":
+                                                          f"Morph_with_Target_{target_file.stem}.cgr"}})
+
+        # Save failed
+        except FileNotFoundError as e:
+            save_fail_message = "\nUnable to create directory"
+            print(save_fail_message)
+            parser.custom_error(str(e))
+
+    # Morph morph_file against all other files in target_directory
+    results = {}
+    for target_file in target_list:
+        if target_file.is_file:
+            # Set the save file destination to be a file within the SLOC directory
+            if save_opt is not None:
+                opts.saveloc = Path(save_morphs_here).joinpath(save_names.get(target_file.name).get("save_morph_as"))
+            # Perform a morph of morph_file against target_file
+            pargs = [morph_file, target_file]
+            results.update({
+                target_file.name:
+                    single_morph(parser, opts, pargs, stdout_flag=False),
+            })
+
+    # Parse rws
+    file_names = []
+    rws = []
+    pearsons = []
+    scales = []
+    stretches = []
+    smears = []
+    results_length = len(results.keys())
+    for key in results.keys():
+        file_names.append(key)
+        rws.append(results[key]["Rw"])
+        pearsons.append(results[key]["Pearson"])
+        if "scale" in results[key]:
+            scales.append(results[key]["scale"])
+        if "stretch" in results[key]:
+            stretches.append(results[key]["stretch"])
+        if "smear" in results[key]:
+            smears.append(results[key]["smear"])
+
+    # Input parameters used for every morph
```

Comment:
I think a header-handling function may also be reusable and can go into another module

Reply:
Put into tools with the name `get_values_from_dictionary_collection()`, though I am not particularly satisfied with that name.